### PR TITLE
feat: support detect pnpm

### PR DESCRIPTION
**Pull request by rxliuli**
When using `pnpm create preact` in a pnpm monorepo, it's necessary to support automatic detection of pnpm for installing dependencies. On the other hand, if within a monorepo, the dependencies should preferably be installed using the package manager employed by the monorepo.

If necessary, I can introduce the testing framework Vitest to add relevant unit tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,4 +30,4 @@
 		"typescript": "^5.0.0"
 	},
 	"prettier": "prettier-config-rschristian"
-}
\ No newline at end of file
+}
```
Comment (rschristian) [inline package.json:33]: Please revert this & the deleted `yarn.lock`

File: src/index.js [hunk 2]
```diff
@@ -16,7 +16,13 @@ const brandColor = /** @type {const} */ ([174, 128, 255]);
...
 	// Don't love the flag, need to find a better name.
 	const skipHint = process.argv.slice(2).includes('--skip-hints');
-	const packageManager = /yarn/.test(process.env.npm_execpath) ? 'yarn' : 'npm';
+	const packageManager =
+		(await detectPackageManager()) ??
+		(/yarn/.test(process.env.npm_execpath)
+			? 'yarn'
+			: process.env.PNPM_PACKAGE_NAME
+			? 'pnpm'
+			: 'npm');
```
Comment (rschristian) [inline src/index.js:25]: Is there any reason why we can't just use `detectPackageManager()` and fallback to NPM?

File: src/index.js [hunk 3]
```diff
@@ -200,26 +206,26 @@ async function templateDir(from, to, useTS) {
 
 /**
  * @param {string} to
- * @param {'yarn' | 'npm'} packageManager
+ * @param {'yarn' | 'npm' | 'pnpm'} packageManager
```
Comment (rschristian) [inline src/index.js:209]: Looks like `@antfu/install-pkg` also supports bun, not that I'm really picky on types, but it'd help to know it's supported.

File: package.json (hunk 1 above)
Comment (rxliuli) [reply package.json]: Sorry, I accidentally deleted it during testing.

File: src/index.js (hunk 2 above)
Comment (rxliuli) [reply src/index.js:25]: `detectPackageManager` does not fall back on checking `process.env` when not in a monorepo, so if the project is not created in a monorepo then it should obviously fall back to the currently used package manager.

not in monorepo

```sh
npm init preact # npm i 
pnpm create preact # pnpm i
yarn create preact # yarn
```

File: src/index.js (hunk 3 above)
Comment (rxliuli) [reply src/index.js]: fixed, https://github.com/preactjs/create-preact/pull/16/commits/06b261820e69e9b18f659a8d5f526f56bf3b0e0c#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R217

File: package.json (hunk 1 above)
Comment (rschristian) [reply package.json]: No worries

Comment (rxliuli) [reply package.json]: fixed, https://github.com/preactjs/create-preact/pull/16/commits/b766b92d97a4e8f044fc27ab504f7b59533c19bd

File: src/index.js (hunk 2 above)
Comment (rschristian) [reply src/index.js:25]: Ah, yeah, weird. Would've assumed that's what they'd have used.

You should be able to use `npm_exectpath` with PNPM too, AFAIK?

Comment (rxliuli) [reply src/index.js:25]: > Ah, yeah, weird. Would've assumed that's what they'd have used.
> 
> You should be able to use `npm_exectpath` 也有 PNPM，据我所知？

No, pnpm is not yet supported. ref: https://github.com/pnpm/pnpm/issues/7037.

Comment (rschristian) [reply src/index.js:25]: Seems to work just fine for me?

> // package.json
```json
{
  "scripts": {
    "test": "echo $npm_execpath"
  }
}
```

Echos the path of the pnpm install as expected.

Comment (rxliuli) [reply src/index.js:25]: Yes, `pnpm run` works, but `pnpm exec` does not.

For instance, you can run `pnpm exec src/index.js`:

```js
// src/index.js
#!/usr/bin/env node
console.log(process.env.npm_execpath)
```

And here's the relevant image illustrating this:

pnpm exec example

Comment (rxliuli) [reply src/index.js:25]: Certainly, perhaps I can first submit a pull request (PR) to pnpm to fix this issue. 😂

Comment (rschristian) [reply src/index.js:25]: Does `pnpm create` fall under `pnpm run` or `pnpm exec`?

I'm clearly not a PNPM user, way too much divergence for my tastes 😅

**Review by rschristian — APPROVED** [review]
Thanks!

Don't worry about the `npm_execpath` thing, was just curious more than anything.

File: src/index.js (hunk 2 above)
Comment (rxliuli) [reply src/index.js:25]: I'm actually not sure if it falls under `pnpm exec`; I just assume that the following commands are more or less equivalent in pnpm.😀

`pnpm create preact` => `pnpm dlx create-preact`(remote download) => `pnpm create-preact`(local) =>  `pnpm exec create-preact`

Comment (rschristian) [reply src/index.js:25]: Looks like `pnpm` does have a `create` sub-command: https://pnpm.io/cli/create

`pnpm create preact` == `yarn create preact` == `npm init preact` (I think)

**Comment by rschristian** [comment]
Apologies for the long delay, should now be available in `0.4.0`

**Pull request merged**